flake-module.nix

```diff
@@ -0,0 +1,230 @@
+{ inputs, lib, self, flake-parts-lib, ... }: with lib; {
+  options = {
+    perSystem = flake-parts-lib.mkPerSystemOption
+      ({ config, options, pkgs, inputs', system, ... }:
+        let
+          cfg = config.terranix;
+        in
+        {
+          options.terranix = {
+
+            setDevShell = mkOption {
+              description = lib.mdDoc ''
+                Whether to set default `devShell` or not. By default enabled.
+                You can disable this if you want to customize the `devShell` creation youself or otherwise
+                wanting to avoid output definition conflicts.
+
+                The devShell are still available at `terranix.terranixConfigurations.<name>.result.devShell`, should
+                you want to incorporate them using `pkgs.mkShell.inputsFrom` or similar.
+              '';
+              type = types.bool;
+              default = true;
+            };
+
+            terranixConfigurations = mkOption
+              {
+                description = lib.mdDoc ''
+                  A submodule of all terranix configurations.'';
+                default = { };
+                type = types.attrsOf
+                  (types.submodule ({ name, ... } @ submod: {
```

Comment:
TIL you can have a default based on the submodule config, I was looking for this in a while and always had to change structure, thank you!